### PR TITLE
Add implementation of ExportCommand to DG

### LINKED ISSUE #245: Documentation: Remove trailing whitespace in UG and DG.
Github actions on repo-wide tests warns on trailing whitespace in UG, DG and PPP.

### LINKED ISSUE #330: DG: Add Implementation Details on Export
This is a non-trivial piece of code and it also strays slightly away from the architecture, perhaps would be good to mention

### PR DESCRIPTION
**Pull request by Bubbl3T**

### CODE REVIEW

**Comment by Bubbl3T** [comment]
Resolves #245.

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -301,74 +279,52 @@ While the `ModuleClass` object for the module "CS1231S" might be saved in `modul
 ```
 
 #### Design considerations
-* **Alternative 1 (Current Choice)**: The `StudentModuleData` and `SessionData` objects only save the `ModuleClass` name and
-`Session` name respectively. Because in any of their functionality, we do not need to know the list of `Sessions` in the
-`ModuleClass` object or the `Date` of the `Session` object. 
+* **Option 1 (Current Choice).**: The `StudentModuleData` and `SessionData` objects only save the `ModuleClass` name and `Session` name respectively. Because in any of their functionality, we do not need to know the list of `Sessions` in the `ModuleClass` object or the `Date` of the `Session` object.
```
Comment (RezwanArefin01) [inline docs/DeveloperGuide.md]: Extra dot after `)`.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -503,18 +440,40 @@ The following sequence diagram shows how changes are propagated to the `UI` thro
...
+  * **Cons**: Complicates design of `Model` as it adds another class just for handling the "view" of a student.
+* **Option 2** Maintain two `ObservableList`-s for `Student` and `SessionData` seperately.
+  * **Pros**: Reduces the need of a encapsulating class, which simplifies the design of `Model`.
+  * **Cons**: Hard to maintain. Accidental slip-up in updating one list but not the other can occur.
+* **Option 3** Let UI query the data from `Student`
+  * **Pros**: `Model` needs no change.
+  * **Cons**: Breaks abstraction principle. Non-trivial querying of fields from `Student` should be handled by `Model`.
+* **Option 4** Pass data through `CommandResult`
+  * **Pros**: Easy to implement.
+  * **Cons**: Hard to maintain. Each `Command` now needs to know that `CommandResult` can pass data other than for result display.
```
Comment (RezwanArefin01) [inline docs/DeveloperGuide.md]: Add `:` after each `Option X`. Also some lines have full stop missing.

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -503,18 +440,40 @@ The following sequence diagram shows how changes are propagated to the `UI` thro
...
+
+### Exporting data as CSV
+This feature allows the user to extract data efficiently from TA-Assist to be used for other purposes such as statistical analysis or result collation on other platforms. Unlike other commands which only works on the model, the `export` command requires access to `Storage` to create a new CSV file and write data to it. This is facilitated by the `ExportCsvStorageAction` class, which is a children of the `StorageAction` class, where further action onto the `Storage` component is processed.
+
+#### CommandResult class
```
Comment (RezwanArefin01) [inline docs/DeveloperGuide.md]: This heading will show up in the ToC. It does not sound like it should be under Exporting implementation details. I would suggest removing this and the next heading.

File: docs/diagrams/ExportCommandSequenceDiagram.puml [hunk 4]
```diff
@@ -0,0 +1,101 @@
+@startuml
+!include style.puml
+
+box Logic LOGIC_COLOR_T1
+participant ":LogicManager" as LogicManager LOGIC_COLOR
+participant ":TaAssistParser" as TaAssistParser LOGIC_COLOR
+participant "e:ExportCommand" as ExportCommand LOGIC_COLOR
+participant ":CommandResult" as CommandResult LOGIC_COLOR
```
Comment (RezwanArefin01) [inline docs/diagrams/ExportCommandSequenceDiagram.puml]: This class is reused in the puml. I think it should be duplicated instead. 
Currently the lifeline of CommandResult doesn't look right in the generated image.

**Review by RezwanArefin01 — CHANGES_REQUESTED** [review]
Minor changes required.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Bubbl3T) [reply docs/DeveloperGuide.md]: Good spot!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Bubbl3T) [reply docs/DeveloperGuide.md]: Sure!

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (Bubbl3T) [reply docs/DeveloperGuide.md]: Sure!

File: docs/diagrams/ExportCommandSequenceDiagram.puml (hunk 4 above)
Comment (Bubbl3T) [reply docs/diagrams/ExportCommandSequenceDiagram.puml]: Fixed in the next commit by creating 2 separate instances of ExportCommand instead.

**Review by RezwanArefin01 — APPROVED** [review]
LGTM!

**Pull request merged**